dm/pkg/encrypt/encrypt.go

```diff
@@ -18,31 +18,61 @@ import (
 	"crypto/aes"
 	"crypto/cipher"
 	"crypto/rand"
-	"encoding/hex"
 
+	"github.com/pingcap/tiflow/dm/pkg/log"
 	"github.com/pingcap/tiflow/dm/pkg/terror"
+	"github.com/pingcap/tiflow/pkg/errors"
 )
 
-var (
-	secretKey, _ = hex.DecodeString("a529b7665997f043a30ac8fadcb51d6aa032c226ab5b7750530b12b8c1a16a48")
```

Comment:
where's the logic that new version DM cluster can decrypt the encrypted password of old version cluster?

Reply:
you mean existing sources/tasks? they're stored in decrypted form actually, dm-worker don't handle encrypt/decrypt.

to upgrade, user should use the same secret key as before to upgrade master first, then worker

Reply:
> you mean existing sources/tasks? they're stored in decrypted form actually, dm-worker don't handle encrypt/decrypt.

I found some `subTaskCfg.DecryptPassword()` call from worker, like https://github.com/pingcap/tiflow/blob/b1c22cfa07e7717e941fd4ccf626bccaf1862c9d/dm/worker/server.go#L931

I'll check if it has no effect soon.